### ISSUE TITLE
Issues with the WT codes

### ISSUE THREAD

**Issue by FatimaPillosu** [esowc/ecPoint-Calibrate#161]
Hi @onyb, 
there is a problem with the WT codes that @Esti, @AugustinVintzileos and @ATimHewson found out yesterday.
As you create the DT, the codes of the WTs change in a funny way. If I have the root with the code 00000, and I split it into two nodes with the code 10000 and 20000, the code for the root changes to 20000. This should not happen.
This problem happens at every node, not only the root. Could you please check this issue? I don't know how this might be influencing the creation of the WTs so I'll put this issue as priority 1.
Cheers,
Fatima

**Comment by onyb** [comment]
**Update:**  This is indeed a bug that I managed to trace to the construction of the decision tree. It's unlikely that this is a regression. I am coming up with an alternative algorithm (using recursion).

According to my assessment, the bug is limited to mislabeled non-leaf nodes and doesn't impact the correctness of mapping functions. I implemented an exhaustive unit-test to ensure the tree construction doesn't break after the proposed fix.
